### PR TITLE
[SR-5856] Diagnostics for protocol extensions store properties

### PR DESCRIPTION
**Pull request by OscarSwanros**
Added a new case for the diagnostics of stored properties on protocol extensions to show a correct error mesgae to the user. 

Resolves [SR-5856](https://bugs.swift.org/browse/SR-5856).

### CODE REVIEW

**Comment by CodaFi** [comment]
@OscarSwanros Please rebase this (remove the compiler changes and leave just the tests).

**Comment by OscarSwanros** [comment]
@CodaFi Done. Thank you!

**Comment by CodaFi** [comment]
Thanks!

Going to manually squash this:

@swift-ci please smoke test

**Comment by CodaFi** [comment]
@swift-ci please smoke test

File: test/decl/var/properties.swift [hunk 1]
```diff
@@ -472,6 +472,18 @@ extension ProtocolWithExtension1 {
...
+protocol ProtocolWithExtension2 {
+  var bar: String { get }
+}
+
+struct StructureImplementingProtocolWithExtension2: ProtocolWithExtension2 {
+  let bar: String
+}
+
+extension ProtocolWithExtension2 {
+  static let baz: Foo = StructureImplementingProtocolWithExtension2(bar: "baz") // expected-error{{static stored properties not supported in protocol extensios}}
```
Comment (CodaFi) [inline test/decl/var/properties.swift]: `Foo` is not a well-defined type

Comment (OscarSwanros) [reply test/decl/var/properties.swift]: Sorry, forgot to rename. Fixed on 69cafb5.

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by jrose-apple** [comment]
Now there are *no* changes?

File: test/decl/var/properties.swift [hunk 2]
```diff
@@ -472,6 +472,18 @@ extension ProtocolWithExtension1 {
...
+protocol ProtocolWithExtension2 {
+  var bar: String { get }
+}
+
+struct StructureImplementingProtocolWithExtension2: ProtocolWithExtension2 {
+  let bar: String
+}
+
+extension ProtocolWithExtension2 {
+  static let baz: ProtocolWithExtension2 = StructureImplementingProtocolWithExtension2(bar: "baz") // expected-error{{static stored properties not supported in protocol extensios}}
```
Comment (jrose-apple) [inline test/decl/var/properties.swift]: Typo: "extensios". Please run tests before you make a pull request! (See docs/Testing.rst for information on how to do this.)

Comment (OscarSwanros) [reply test/decl/var/properties.swift]: Alright, sorry, was mobile.

**Comment by jrose-apple** [comment]
Sorry for the overly terse comments! The main thing is that your merge seems to have wiped out your changes. (The changes themselves look good.)

**Comment by OscarSwanros** [comment]
No worries, @CodaFi merged #11930 which was submitted minutes before mine and was the exact same thing. This one just adds tests.

**Comment by CodaFi** [comment]
CI status didn't reset when you rebased 😳

**Comment by jrose-apple** [comment]
Oh, I'm sorry for missing that context!

**Comment by jrose-apple** [comment]
@swift-ci please smoke test

**Comment by jrose-apple** [comment]
Hm, clearly unrelated to this commit:

```
src/libdispatch.so: error: undefined reference to '__atomic_fetch_sub_4'
src/libdispatch.so: error: undefined reference to '__atomic_load_4'
src/libdispatch.so: error: undefined reference to '__atomic_fetch_add_4'
src/libdispatch.so: error: undefined reference to '__atomic_compare_exchange_4'
src/libdispatch.so: error: undefined reference to '__atomic_fetch_or_4'
src/libdispatch.so: error: undefined reference to '__atomic_fetch_and_4'
src/libdispatch.so: error: undefined reference to '__atomic_store_4'
src/libdispatch.so: error: undefined reference to '__atomic_exchange_4'
```

@dgrove-oss, any ideas?

**Comment by dgrove-oss** [comment]
would suspect change in compiler version on buildbots causing some problem with how the intrinsics for c11_atomics are named.   Looks like it worked itself out in later full builds?

**Comment by jrose-apple** [comment]
@swift-ci please smoke test

**Comment by jrose-apple** [comment]
@swift-ci Please smoke test Linux

**Comment by OscarSwanros** [comment]
Thanks!

**Pull request merged**